Backend/utils/cronJobs.js

```diff
@@ -1,8 +1,326 @@
+                                genres: [],
+                            }
+                        });
+                    }
+                }
+            } else {
+                console.error(`Failed to fetch tracks for playlist ${playlist.id}:`, data);
+            }
+        } catch (error) {
+            console.error('Error fetching tracks:', error);
+        }
+
+        await delay(1000); // Add a delay between playlist fetches to reduce rate limiting
+    }
+}
+
+async function fetchAndStoreArtistGenres() {
+    const artists = await prisma.artist.findMany(); // Fetch all artists, not just those with empty genres
+
+    const accessToken = await getSystemUserToken();
+
+    for (const artist of artists) {
+        try {
+
+            const response = await fetchWithRetry(
+                `https://api.spotify.com/v1/artists/${artist.spotifyId}`,
+                {
+                    headers: {
+                        'Authorization': `Bearer ${accessToken}`
+                    }
+                }
+            );
+
+            if (!response.ok) {
+                const errorText = await response.text();
+                console.error(`Failed to fetch genres for artist ${artist.spotifyId}:`, errorText);
+                await delay(1000); // Delay for 1 second before the next request
+                continue; // Skip this artist and continue with the next
+            }
+
+            const data = await response.json();
+
+            await prisma.artist.update({
+                where: { spotifyId: artist.spotifyId },
+                data: { genres: data.genres }
+            });
+
+            // Ensure getLocationForArtist is called
+            const location = await getLocationForArtist(artist);
+
+            if (location && location.name) { // Check if location is valid
+                await prisma.location.upsert({
+                    where: { name: location.name },
+                    update: {
+                        artists: {
+                            connect: { id: artist.id }
+                        }
+                    },
+                    create: {
+                        name: location.name,
+                        latitude: location.latitude,
+                        longitude: location.longitude,
+                        countryCode: location.countryCode,
+                        artists: {
+                            connect: { id: artist.id }
+                        }
+                    }
+                });
+            } else {
+                console.warn(`No valid location found for artist ${artist.name}. Skipping location update.`);
+            }
+        } catch (error) {
+            console.error(`Error fetching genres for artist ${artist.name}:`, error);
+        }
+
+        await delay(1000); // Delay for 1 second before the next request
+    }
+
+}
+
+cron.schedule('0 * * * *', async () => {
+    console.log('Cron job started: Running cron job to fetch Spotify data...');
```

Comment:
We don't want to commit console.log statements - remove them from this file. If you really want to keep them temporarily for debugging purposes, maybe you can add a comment with "TODO: cleanup" above so you remember to remove them later

Reply:
Got it! Made the changes in the "trending-algorithm" branch's commit though, to avoid merge conflicts.